### PR TITLE
Add a statistics endpoint for the ocr-api

### PR DESCRIPTION
**Pull request by SteveHicksCH**
Jira: IVP-1369

This allows clients (support) to query the ocr-api for the state of the service. The key statistic is the amount of messages waiting on an internal queue

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/ocr/api/statistics/StatisticsService.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    private static final String INSTANCE_UUID = UUID.randomUUID().toString();
+
+    @Autowired
+    private ThreadConfig threadConfig;
+
+    @Autowired
+    private ThreadPoolTaskExecutor taskExecutor;
+
+    public StatisticsDTO create() {
```
Comment (csciubbach) [inline src/main/java/uk/gov/companieshouse/ocr/api/statistics/StatisticsService.java:22]: Should rename this to something like generateStatistics maybe

File: src/main/java/uk/gov/companieshouse/ocr/api/statistics/StatisticsDTO.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package uk.gov.companieshouse.ocr.api.statistics;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class StatisticsDTO {
```
Comment (csciubbach) [inline src/main/java/uk/gov/companieshouse/ocr/api/statistics/StatisticsDTO.java:5]: Should be StatisticsDto not StatisticsDTO

File: src/test/java/uk/gov/companieshouse/ocr/api/image/extracttext/ImageOcrApiControllerTest.java [hunk 3]
```diff
@@ -31,7 +31,7 @@
 @Tag(TestType.UNIT)
 @ExtendWith(SpringExtension.class)
 @WebMvcTest(controllers = ImageOcrApiController.class)
-public class ImageOcrApiControllerTest {
+class ImageOcrApiControllerTest {
```
Comment (csciubbach) [inline src/test/java/uk/gov/companieshouse/ocr/api/image/extracttext/ImageOcrApiControllerTest.java:34]: add public back. Better explicit. Some languages have opposite defaults. Always good to include accessor level than rely on default. Probably never going to happen but imagine they changed the default?

Comment (csciubbach) [reply src/test/java/uk/gov/companieshouse/ocr/api/image/extracttext/ImageOcrApiControllerTest.java:34]: This comment applies to all the other ones...

File: src/test/java/uk/gov/companieshouse/ocr/api/image/extracttext/TextConversionResultTest.java [hunk 4]
```diff
@@ -26,7 +26,7 @@ public void setup() {
     }
 
     @Test
-    public void singlePageResults5Datapoints() {
```
Comment (csciubbach) [inline src/test/java/uk/gov/companieshouse/ocr/api/image/extracttext/TextConversionResultTest.java:29]: Make these public

File: src/main/java/uk/gov/companieshouse/ocr/api/statistics/StatisticsService.java (hunk 1 above)
Comment (SteveHicksCH) [reply src/main/java/uk/gov/companieshouse/ocr/api/statistics/StatisticsService.java:22]: Agreed

File: src/test/java/uk/gov/companieshouse/ocr/api/image/extracttext/TextConversionResultTest.java (hunk 4 above)
Comment (SteveHicksCH) [reply src/test/java/uk/gov/companieshouse/ocr/api/image/extracttext/TextConversionResultTest.java:29]: That gives Sonar errors that the lead developers don't want

File: src/test/java/uk/gov/companieshouse/ocr/api/image/extracttext/ImageOcrApiControllerTest.java (hunk 3 above)
Comment (SteveHicksCH) [reply src/test/java/uk/gov/companieshouse/ocr/api/image/extracttext/ImageOcrApiControllerTest.java:34]: That gives Sonar errors that the lead developers don't want

File: src/main/java/uk/gov/companieshouse/ocr/api/statistics/StatisticsDTO.java (hunk 2 above)
Comment (SteveHicksCH) [reply src/main/java/uk/gov/companieshouse/ocr/api/statistics/StatisticsDTO.java:5]: Updated

File: src/test/java/uk/gov/companieshouse/ocr/api/image/extracttext/TextConversionResultTest.java (hunk 4 above)
Comment (csciubbach) [reply src/test/java/uk/gov/companieshouse/ocr/api/image/extracttext/TextConversionResultTest.java:29]: Sonar is wrong on this, we need to disable the rule

**Review by csciubbach — APPROVED** [review]

**Pull request merged**